ibm/service/appconfiguration/data_source_ibm_app_config_segment.go

```diff
@@ -0,0 +1,265 @@
+			},
+			"created_time": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Creation time of the segment.",
+			},
+			"updated_time": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Last modified time of the segment data.",
+			},
+			"href": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Segment flag URL.",
+			},
+			"rules": {
+				Type:        schema.TypeList,
+				Computed:    true,
+				Description: "List of rules that determine if the entity belongs to the segment during feature / property evaluation.",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"values": {
+							Type:        schema.TypeList,
+							Computed:    true,
+							Description: "List of values. Entities matching any of the given values will be considered to belong to the segment",
+							Elem: &schema.Schema{
+								Type: schema.TypeString,
+							},
+						},
+						"attribute_name": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "Attribute name.",
+						},
+						"operator": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "Operator to be used for the evaluation if the entity belongs to the segment.",
+						},
+					},
+				},
+			},
+			"features": {
+				Type:        schema.TypeList,
+				Computed:    true,
+				Description: "List of Features associated with the segment.",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"feature_id": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "feature id.",
+						},
+						"name": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "Name of the Feature.",
+						},
+					},
+				},
+			},
+			"properties": {
+				Type:        schema.TypeList,
+				Computed:    true,
+				Description: "List of Features associated with the segment.",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"property_id": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "property id.",
+						},
+						"name": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "Name of the Property.",
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func dataSourceIbmAppConfigSegmentRead(d *schema.ResourceData, meta interface{}) error {
+	guid := d.Get("guid").(string)
+
+	appconfigClient, err := getAppConfigClient(meta, guid)
+	if err != nil {
+		return err
+	}
+
+	options := &appconfigurationv1.GetSegmentOptions{}
+	options.SetSegmentID(d.Get("segment_id").(string))
+
+	if _, ok := d.GetOk("includes"); ok {
+		includes := []string{}
+		for _, segmentsItem := range d.Get("includes").([]interface{}) {
+			includes = append(includes, segmentsItem.(string))
+		}
+		options.SetInclude(includes)
+	}
+
```

Comment:
There are many other optional arguments like name, description, tags that are not included in the payload..
If arguments are not used in request payload, please mark it as computed

Reply:
Addressed